opt/scopeserver/dataserver/utils/loom_file_handler.py

```diff
@@ -80,71 +57,47 @@ def change_loom_mode(
-                logger.debug(e)
-                pass
         return self.active_looms[abs_file_path].get_connection()
 
-    def get_loom_absolute_file_path(self, loom_file_path: str) -> str:
-        return os.path.join(self.loom_dir, loom_file_path)
+    def get_loom_absolute_file_path(self, loom_file_path: Path) -> Path:
+        return self.loom_dir / loom_file_path
 
     def get_global_looms(self) -> list:
         return self.global_looms
 
     def set_global_data(self) -> None:
         self.global_looms = [x for x in os.listdir(self.loom_dir) if not os.path.isdir(os.path.join(self.loom_dir, x))]
 
-    def get_loom_connection(self, loom_file_path: str, mode: str = "r"):
+    def get_loom_connection(self, loom_file_path: Path, mode: str = "r") -> LoomConnection:
         logger.debug(f"Getting connection for {loom_file_path} in mode {mode}")
         return self.get_loom(loom_file_path=loom_file_path, mode=mode).get_connection()
 
-    def get_loom(self, loom_file_path: str, mode: str = "r"):
+    def get_loom(self, loom_file_path: Path, mode: str = "r") -> Loom:
         abs_loom_file_path = self.get_loom_absolute_file_path(loom_file_path)
-        if not os.path.exists(abs_loom_file_path):
-            logger.error(f"The file {loom_file_path} does not exists.")
-            raise ValueError("The file located at " + abs_loom_file_path + " does not exist.")
-        # To check if the given file path is given specified url!
-        partial_md5_hash = LoomFileHandler.get_partial_md5_hash(abs_loom_file_path, 10000)
-        if abs_loom_file_path in self.active_looms:
-            logger.debug("Should be preloaded")
-            try:
-                logger.debug(
-                    f"Current mode: {self.active_looms[abs_loom_file_path].get_connection().mode}, wanted mode {mode}"
-                )
-
-                if self.active_looms[abs_loom_file_path].get_connection().mode == mode:
-                    loom = self.active_looms[abs_loom_file_path]
+        with self.file_locks[abs_loom_file_path]:
+            if not os.path.exists(abs_loom_file_path):
+                logger.error(f"The file {loom_file_path} does not exists.")
+                raise ValueError(f"The file located at {abs_loom_file_path} does not exist.")
+            if abs_loom_file_path in self.active_looms:
+                logger.debug("Should be preloaded")
+                loom = self.active_looms[abs_loom_file_path]
+                try:
                     logger.debug(
-                        f"Returning pre-loaded loom file {loom_file_path}. Hash {partial_md5_hash}, object {id(loom)}"
+                        f"Current mode: {self.active_looms[abs_loom_file_path].get_connection().mode}, wanted mode {mode}"
                     )
-                    return loom
-            except AttributeError:
-                logger.error("Loom was previously closed")
+                    if self.active_looms[abs_loom_file_path].get_connection().mode == mode:
```

Comment:
What happens if the `mode` does not match? The `try`/`catch` falls through and you return the loom at the end of the function

Reply:
Yeah. I;'m not sure if theres something better here, we should at least log the error. If we don't return the loom, the frontend could end up hanging somewhere, but we don't have feedback to the user that something failed.

Reply:
As discussed on Slack, this will be done in a separate PR for full integration of the Result class.